### PR TITLE
Transmission BitTorrent client completions

### PR DESCRIPTION
**Pull request by aftix**
This PR adds completions for the transmission BitTorrent client. The included commands are:

- `transmission-cli`
- `transmission-edit`
- `transmission-show`
- `transmission-create`
- `transmission-daemon`
- `transmission-remote`

Some of these commands have tab completions for IP addresses, so I experimented with adding an action for IPv6 to match the current IPv4 action, but decided that each segment of an IPv6 address has too many options and the completions just fill the terminal.

### CODE REVIEW

**Review by rsteube — CHANGES_REQUESTED** [review]
`ActionIds` is the only blocking issue.

File: completers/transmission-edit_completer/cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	rootCmd.Flags().StringP("add", "a", "", "Add an announce URL to the torrent's announce-list")
+	rootCmd.Flags().StringP("delete", "d", "", "Remove an announce URL from the torrent's announce-list")
+	rootCmd.Flags().BoolP("help", "h", false, "Show a short help page and exit")
+	rootCmd.Flags().StringP("replace", "r", "", "Substring search-and-replace inside a torrent's announce-list URLs")
+	rootCmd.Flags().StringP("source", "s", "", "Sets the source tag within a torrent")
+	rootCmd.Flags().BoolP("version", "V", false, "Show version number and exit")
+
+	carapace.Gen(rootCmd).FlagCompletion(carapace.ActionMap{
+		"replace": carapace.ActionMultiParts(" ", func(c carapace.Context) carapace.Action {
+			return carapace.ActionValues()
```
[the remark this answers is not in the document]
Comment (aftix) [reply completers/transmission-edit_completer/cmd/root.go:31]: It isn't, how do I encode that --replace is followed by two strings separated by a space?

[the remark this answers is not in the document]
Comment (rsteube) [reply completers/transmission-edit_completer/cmd/root.go:31]: Ah, it's one of those abominations.

I got basic support for this now in [pflag](https://github.com/carapace-sh/carapace-pflag?tab=readme-ov-file#nargs):
```go
rootCmd.Flag("replace").Nargs = 2
```

**Comment by aftix** [comment]
Updated the branch to be rebased against master again and pushed a commit addressing all the review comments

**Comment by rsteube** [comment]
thanks

**Pull request merged**